src/cpl/share_esmf/lnd_set_decomp_and_domain.F90

```diff
@@ -0,0 +1,881 @@
+          lon1d(n) = rtemp_glob(n)
+       end do
+
+       ! Determine lat1d
+       allocate(lndlats_glob(gsize))
+       lndlats_glob(:) = 0._r8
+       do n = 1,numownedelements
+          lndlats_glob(gindex(n)) = ownedElemCoords(2*n)
+       end do
+       call ESMF_VMAllReduce(vm, sendData=lndlats_glob, recvData=rtemp_glob, count=gsize, &
+            reduceflag=ESMF_REDUCE_SUM, rc=rc)
+       deallocate(lndlats_glob)
+       allocate(lat1d(nj))
+       do n = 1,nj
+          lat1d(n) = rtemp_glob((n-1)*ni + 1)
+       end do
+       deallocate(rtemp_glob)
+    end if
+
+    ! Sanity check- remove this when it is done
+    call getfil( trim(fatmlndfrc), locfn, 0 )
+    call ncd_pio_openfile (ncid, trim(locfn), 0)
+    allocate(lonc_atmlndfrc(numownedelements))
+    allocate(latc_atmlndfrc(numownedelements))
+    call ncd_io(ncid=ncid, varname= 'xc' , flag='read', data=lonc_atmlndfrc , dim1name=grlnd)
+    call ncd_io(ncid=ncid, varname= 'yc' , flag='read', data=latc_atmlndfrc , dim1name=grlnd)
+    do g = begg,endg
+       n = g - begg + 1
+       if ( abs(lonc_atmlndfrc(n) - ldomain%lonc(g)) > 1.e-11 .and. &
+            abs(lonc_atmlndfrc(n) - ldomain%lonc(g)) /= 360._r8) then
+          write(6,'(a,3(d20.13,2x))')'ERROR: lonc_atmlndfrac(n), ldomain%lonc(g), abs(diff) = ',&
+               lonc_atmlndfrc(n), ldomain%lonc(g), abs(lonc_atmlndfrc(n) - ldomain%lonc(g))
+          call shr_sys_abort()
+       end if
+       if (abs(latc_atmlndfrc(n) - ldomain%latc(g)) > 1.e-11) then
+          write(6,'(a,3(d20.13,2x))')'ERROR: latc_atmlndfrac(n), ldomain%latc(g), abs(diff) = ',&
+               latc_atmlndfrc(n), ldomain%latc(g), abs(latc_atmlndfrc(n) - ldomain%latc(g))
+          call shr_sys_abort()
+       end if
+    end do
+    deallocate(lonc_atmlndfrc)
+    deallocate(latc_atmlndfrc)
+    call ncd_pio_closefile(ncid)
+
+  end subroutine lnd_set_ldomain_gridinfo_from_mesh
+
+  !===============================================================================
+  subroutine pio_check_err(ierror, description)
+    use pio, only : PIO_NOERR
+    integer     , intent(in) :: ierror
+    character(*), intent(in) :: description
+    if (ierror /= PIO_NOERR) then
+       write (*,'(6a)') 'ERROR ', trim(description)
+       call shr_sys_abort()
+    endif
+  end subroutine pio_check_err
+
+  !===============================================================================
+  logical function chkerr(rc, line, file)
+    integer          , intent(in) :: rc
+    integer          , intent(in) :: line
+    character(len=*) , intent(in) :: file
+
+    integer :: lrc
+    chkerr = .false.
+    lrc = rc
+    if (ESMF_LogFoundError(rcToCheck=lrc, msg=ESMF_LOGERR_PASSTHRU, line=line, file=file)) then
+       chkerr = .true.
+    endif
+  end function chkerr
+
+  !===============================================================================
+  subroutine lnd_set_read_write_landmask(write_file, read_file, lndmask_glob, lndfrac_glob, gsize)
```

Comment:
- [ ] This subroutine appears not to be used. Can I remove it, or do you need it for your upcoming branch?

Reply:
Please go ahead and remove it.

Reply:
I added this in case there were problems with the generation of a land mask that was different with different number of tasks. I have not found this to be the case. But for now I would keep it in and add a comment that this is not currently being used but is a place holder in case we do need to use it in the future. Does that make sense?